src/util.rs

```diff
@@ -149,3 +149,48 @@ mod tests {
         }
     }
 }
+
+#[cfg(kani)]
+mod proofs {
+    use super::*;
+
+    #[kani::proof]
+    fn prove_round_down_to_next_multiple_of_alignment() {
+        fn model_impl(n: usize, align: NonZeroUsize) -> usize {
+            debug_assert!(align.get().is_power_of_two());
```

Comment:
Since we're in tests, might as well make this a non-debug assert:

```suggestion
            assert!(align.get().is_power_of_two());
```